File: RoomDb/app/src/main/java/com/example/roomdb/MainActivity.java
package com.example.roomdb;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button btProfesor, btCurso, btLenguajes, btProfesorLenguaje;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        configView();
    }
    private void configView(){
        btProfesor = findViewById(R.id.mainAcitivityBtProfessor);
        btProfesor.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getApplicationContext(), ProfessorActivity.class));
            }
        });
        btCurso = findViewById(R.id.mainAcitivityBtCurso);
        btCurso.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getApplicationContext(), CourseActivity.class));
            }
        });

        btLenguajes = findViewById(R.id.mainAcitivityBtLenguajes);
        btLenguajes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getApplicationContext(), LanguagesActivity.class));
            }
        });
    }
}

File: Retrofit/app/src/main/java/com/example/retrofit/MainActivity.kt
package com.example.retrofit

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.TextView
import android.widget.Toast

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

import retrofit2.Retrofit
import retrofit2.awaitResponse
import retrofit2.converter.gson.GsonConverterFactory
const val BASE_URL = "https://cat-fact.herokuapp.com"
val TAG = "MainActivity"

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        getCurrentData()

        findViewById<Button>(R.id.button).setOnClickListener{
            getCurrentData()
        }

    }

    private fun getCurrentData() {
        findViewById<TextView>(R.id.tv_textView).visibility = View.GONE
        findViewById<TextView>(R.id.tv_timeStamp).visibility = View.GONE
        findViewById<ProgressBar>(R.id.progressBar).visibility = View.VISIBLE

        val api = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(ApiInterface::class.java)

        GlobalScope.launch(Dispatchers.IO) {
            try {
                val response = api.getCatFacts().awaitResponse()
                if (response.isSuccessful) {

                    val data = response.body()!!
                    Log.d(TAG, data.toString())

                    withContext(Dispatchers.Main) {
                        findViewById<TextView>(R.id.tv_textView).visibility = View.VISIBLE
                        findViewById<TextView>(R.id.tv_timeStamp).visibility = View.VISIBLE
                        findViewById<ProgressBar>(R.id.progressBar).visibility = View.GONE
                        findViewById<TextView>(R.id.tv_timeStamp).text = data.createdAt
                        findViewById<TextView>(R.id.tv_textView).text = data.text

                    }

                }
            } catch (e: Exception) {
                withContext(Dispatchers.Main){
                    Toast.makeText(
                        applicationContext,
                        "Seems like something went wrong...",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
        }


    }
}